geospatial/src/Dem.cc

```diff
@@ -275,35 +284,78 @@ float Dem::MaxElevation() const
 bool Dem::GeoReference(double _x, double _y,
     gz::math::Angle &_latitude, gz::math::Angle &_longitude) const
 {
-  if (this->dataPtr->isNonEarthDem)
+  if (this->dataPtr->isUnknownDem)
   {
-    gzerr << "Can not retrieve WGS84 coordinates from non-Earth DEM."
+    gzerr << "Can not retrieve coordinates from unknown DEM."
             << std::endl;
     return false;
   }
 
   double geoTransf[6];
   if (this->dataPtr->dataSet->GetGeoTransform(geoTransf) == CE_None)
   {
-    OGRSpatialReference sourceCs;
-    OGRSpatialReference targetCs;
-    OGRCoordinateTransformation *cT;
+    OGRCoordinateTransformation *cT = nullptr;
     double xGeoDeg, yGeoDeg;
 
-    // Transform the terrain's coordinate system to WGS84
-    const char *importString
-        = strdup(this->dataPtr->dataSet->GetProjectionRef());
-    if (importString == nullptr || importString[0] == '\0')
+    if (this->dataPtr->sphericalCoordinates.Surface() ==
+        math::SphericalCoordinates::EARTH_WGS84)
     {
-      gzdbg << "Projection coordinate system undefined." << std::endl;
-      return false;
+      OGRSpatialReference sourceCs;
+      OGRSpatialReference targetCs;
+      // Transform the terrain's coordinate system to WGS84
+      const char *importString
+          = strdup(this->dataPtr->dataSet->GetProjectionRef());
+      if (importString == nullptr || importString[0] == '\0')
+      {
+        gzdbg << "Projection coordinate system undefined." << std::endl;
+        return false;
```

Comment:
Is it possible to add a test for this to increase code coverage?

Reply:
So the importString is taken from the DEM file, and this section would only run when that is a ``nullptr``, so I'm not sure how to make a new DEM file with that information removed.

Reply:
Ah ok, let's have coverage skip it with:
```suggestion
        // LCOV_EXCL_START
        gzdbg << "Projection coordinate system undefined." << std::endl;
        return false;
        // LCOV_EXCL_STOP
```